### ISSUE TITLE
Mouse click at bottom of 300 line file puts cursor there, but the viewport at the last position.

### ISSUE THREAD

**Issue by raguay** [onivim/oni2#498]
Open a large file (300 lines in my case), set the cursor at the top, scroll to the bottom and mouse click the line. The cursor point will be placed properly, but the viewport scrolls back to the last position of the cursor. Very reproducible.

**Comment by erickzanardo** [comment]
I can see this happening even with smaller files.

**Comment by raguay** [comment]
I just downloaded the latest version and it has this issue much worse! Now, the viewport gets stuck in a particular location. Moving the cursor with j/k to off the viewport will not scroll the viewport. If I scroll the viewport the edit/move the cursor with keyboard commands, the viewport goes back to the stuck position and the cursor is still on the line I wanted to edit, but I now can't see it.

When it gets stuck, I can do `:wall` a few times and it gets unstuck. But, it will get stuck again fairly quickly. I haven't been able to determine the action that gets the viewport stuck at one place.

The file just has to be larger than the viewing area. I was editing a 300 line file when I first noticed the issue.
